aepsych/models/inducing_point_allocators.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+import torch
+
+from aepsych.config import Config, ConfigurableMixin
+from aepsych.utils import get_bounds
+from botorch.models.utils.inducing_point_allocators import InducingPointAllocator
+from botorch.utils.sampling import draw_sobol_samples
+from scipy.cluster.vq import kmeans2
+
+
+class SobolAllocator(InducingPointAllocator, ConfigurableMixin):
+    """An inducing point allocator that uses Sobol sequences to allocate inducing points."""
+
+    def __init__(self, bounds) -> None:
+        """Initialize the SobolAllocator without bounds."""
+        self.bounds = bounds
+        super().__init__()
+
+    def _get_quality_function(self) -> None:
+        """Sobol sampling does not require a quality function, so this returns None."""
+        return None
+
+    def allocate_inducing_points(
+        self,
+        inputs: Optional[torch.Tensor] = None,
+        covar_module: Optional[torch.nn.Module] = None,
+        num_inducing: int = 10,
+        input_batch_shape: torch.Size = torch.Size([]),
+    ) -> torch.Tensor:
+        """
+        Generates `num_inducing` inducing points within the specified bounds using Sobol sampling.
+
+        Args:
+            inputs (torch.Tensor): Input tensor, not required for Sobol sampling.
+            covar_module (torch.nn.Module, optional): Kernel covariance module; included for API compatibility, but not used here.
+            num_inducing (int, optional): The number of inducing points to generate. Defaults to 10.
+            input_batch_shape (torch.Size, optional): Batch shape, defaults to an empty size; included for API compatibility, but not used here.
+
+
+        Returns:
+            torch.Tensor: A (num_inducing, d)-dimensional tensor of inducing points within the specified bounds.
+
+        Raises:
+            ValueError: If `bounds` is not provided.
+        """
+
+        # Validate bounds shape
+        assert (
+            self.bounds.shape[0] == 2
+        ), "Bounds must have shape (2, d) for Sobol sampling."
+
+        # Generate Sobol samples within the unit cube [0,1]^d and rescale to [bounds[0], bounds[1]]
+        inducing_points = draw_sobol_samples(
+            bounds=self.bounds, n=num_inducing, q=1
+        ).squeeze()
+
+        # Ensure correct shape in case Sobol sampling returns a 1D tensor
+        if inducing_points.ndim == 1:
+            inducing_points = inducing_points.view(-1, 1)
+
+        return inducing_points
+
+    @classmethod
+    def from_config(
```

Comment:
I think we can remove this, and all the other `from_config` methods, because these classes can just inherit the base implementation from the mixin, no?